src/Microsoft.Extensions.Configuration.AzureAppConfiguration/CdnApiVersionPolicy.cs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using Azure.Core;
+using Azure.Core.Pipeline;
+using System;
+using System.Collections.Specialized;
+using System.Threading.Tasks;
+using System.Web;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration
+{
+    /// <summary>
+    /// A policy that adds the API version query parameter to HTTP requests.
+    /// </summary>
+    public class CdnApiVersionPolicy : HttpPipelinePolicy
+    {
+        /// <summary>
+        /// Processes the HTTP message by adding the API version query parameter.
+        /// </summary>
+        /// <param name="message">The HTTP message to process.</param>
+        /// <param name="pipeline">The pipeline of HTTP policies to apply.</param>
+        public override void Process(HttpMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline)
+        {
+            message.Request.Uri.Reset(AlterApiVersion(message.Request.Uri.ToUri()));
```

Comment:
Seems like we can follow the pattern used in [`UserAgentHeaderPolicy`](https://github.com/Azure/AppConfiguration-DotnetProvider/blob/90835cf5efbfb26d6de6068e4c31c3050947b78a/src/Microsoft.Extensions.Configuration.AzureAppConfiguration/UserAgentHeaderPolicy.cs#L17) so we don't have to repeat the same logic in `Process`/`ProcessAsync`

Reply:
In order to not repeat one line (which I consider completely fine), I would need to add another wrapper method that is basically an if-else statement with a boolean.

I would say that pattern is overkill and not the simplest solution